HTTP_HEADER_FORMAT.md

```diff
@@ -61,26 +60,97 @@ caller rather than strict rules to follow for 3 reasons:
 3.  Different load between caller service and callee service might force callee
     to down sample.
 
-The behavior of other bits is currently undefined.
+### Prefix '.' The Hierarchical-id
 
-#### Examples of HTTP headers
+The '.' character indicates that a variable number of characters follow and represent
+a hierarchical (path based) id.   It has the form of a list of hexadecimal numbers 
+(lower case) separated by '.' characters.   The list is optionally terminated by 
+a '!' character.   The id ends if either the string ends or a character that 
+is not a hexadecimal character or a '.' or '!' is encountered.   
 
-*Valid sampled Trace-Context:*
+Examples
 
 ```
-Value = 00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01
-base16(<Version>) = 00
-base16(<TraceId>) = 4bf92f3577b34da6a3ce929d0e0e4736
-base16(<SpanId>) = 00f067aa0ba902b7
-base16(<TraceOptions>) = 01  // sampled
+	.1
+	.3.b.34.ca25.56
+	.334.ba34.93a4.1.1!
 ```
+The hierarchical ID is meant to represent series of causal requests within the scope
+of a given Trace-id.   Take has a whole, like the Span-ID, combined with the Trace-id
+it uniquely identifies the request.    However in addition the components also represent
+other requests with that same trace that are causality connects.  For example the Trace-Context
+value of 
+```
+*4bf92f3577b34da6a3ce929d0e0e4736.3.b.34
+```
+Represents a request with ID
+```
+*4bf92f3577b34da6a3ce929d0e0e4736.3.b.34
+```
+that was caused by a request with ID
+```
+*4bf92f3577b34da6a3ce929d0e0e4736.3.b
+```
+which in turn was caused by request with 
+```
+*4bf92f3577b34da6a3ce929d0e0e4736.3
+```
+Hierarchical ID are conceptually unbounded in length, but as a practical matter must be
+limited, if only to avoid issues when infinite request loops form.   The '!' character 
+is meant to indicate the ID has been truncated.  The ID as a whole should still be unique
+within the Trace-id, but the fine grained causality information will not be present.   
+
+### Examples of HTTP headers
 
-*Valid not-sampled Trace-Context:*
+Note that not all the allowed value prefixes are expected to be used simultaneously.  __In particular it is expected that only one of the Span-id or Hierarchical-id will be used.__
```

Comment:
While generally this is true, we will use the span-id in conjunction with the hierarchical-id when we reset the hierarchical-id, to handle bad callers. When reset, a new span-id is generated and the hierarchical-id could be generated with subsequent extensions and increments.